tools/sdk-sync/src/sync.rs

```diff
@@ -0,0 +1,685 @@
+            // by the end of this loop.
+            self.smithy_rs
+                .hard_reset(commit.hash.as_ref())
+                .with_context(|| format!("failed to reset to {} in smithy-rs", rev,))?;
+
+            self.copy_original_examples().context(here!())?;
+            self.build_and_copy_sdk(&versions.aws_doc_sdk_examples_revision)
+                .context("failed to generate the SDK")?;
+            self.commit_sdk_changes("[smithy-rs]", &commit)
+                .context("couldn't commit SDK changes to aws-sdk-rust")?;
+        }
+
+        info!(
+            "Successfully synced {} smithy-rs commit(s) to aws-sdk-rust",
+            commits.len(),
+        );
+
+        Ok(())
+    }
+
+    /// Aggregate the commits made to Rust SDK examples into one while maintaining attribution.
+    /// We squash all the example commits since the examples repo does merges rather than squash merges,
+    /// and we prefer squash merges in aws-sdk-rust.
+    #[instrument(skip(self, versions))]
+    fn sync_examples(&self, versions: &VersionsManifest) -> Result<()> {
+        // Only consider revisions on the `rust_dev_preview/` directory since
+        // copying over changes to other language examples is pointless
+        let example_revisions = self
+            .aws_doc_sdk_examples
+            .rev_list(
+                "HEAD",
+                versions.aws_doc_sdk_examples_revision.as_ref(),
+                Some(&PathBuf::from("rust_dev_preview")),
+            )
+            .context(here!())?;
+        if example_revisions.is_empty() {
+            info!("No example changes to copy over.");
+            return Ok(());
+        }
+        let examples_head = example_revisions.iter().cloned().next().unwrap();
+
+        let from = self.aws_doc_sdk_examples.path().join("rust_dev_preview");
+
+        info!("Cleaning examples...");
+        self.fs
+            .remove_dir_all_idempotent(&self.smithy_rs.path().join("aws/sdk/examples"))
+            .context(here!())?;
+
+        info!(
+            "Copying examples from {:?} to 'smithy-rs/aws/sdk/examples'...",
+            from
+        );
+        self.fs
+            .recursive_copy(&from, &self.smithy_rs.path().join("aws/sdk/examples"))
+            .context(here!())?;
+        self.fs
+            .remove_dir_all_idempotent(&self.smithy_rs.path().join("aws/sdk/examples/.cargo"))
+            .context(here!())?;
+        self.fs
+            .remove_file(&self.smithy_rs.path().join("aws/sdk/examples/Cargo.toml"))
+            .context(here!())?;
+
+        self.build_and_copy_sdk(&examples_head).context(here!())?;
+        self.aws_sdk_rust
+            .stage(&PathBuf::from("."))
+            .context(here!())?;
+
+        let example_commits: Vec<Commit> = example_revisions
+            .iter()
+            .map(|commit_hash| self.aws_doc_sdk_examples.show(commit_hash.as_ref()))
+            .collect::<Result<_, _>>()?;
```

Comment:
I've never seen both the type annotation AND a turbofish but it totally makes sense in this case.